File: HackerRank/Challenges/GameOfThrones1.py
#!/bin/python3

import math
import os
import random
import re
import sys
from collections import Counter

# Complete the gameOfThrones function below.


def gameOfThrones(s):
    table = Counter()
    for i in s:
        table[i] += 1
    count = 0
    for i in table:
      if table[i]%2!=0:
        count+=1
      if count>1:
        break
    if count>1:
      return "NO"
    return "YES"


if __name__ == '__main__':
    # fptr = open(os.environ['OUTPUT_PATH'], 'w')

    s = input()

    result = gameOfThrones(s)
    print(result)

    # fptr.write(result + '\n')

    # fptr.close()

File: HackerRank/Challenges/GameOfThrones1.js
// let s = "aaabbbb";
let s = "cdefghmnopqrstuvw";
function gameOfThrones(s) {
  let table = {};
  for (let i of s) {
    table[i] = (table[i] || 0) + 1;
  }
  let count = 0;
  for (let i of Object.values(table)) {
    if (i % 2 !== 0) count++;
    if (count > 1) break;
  }
  if (count > 1) return "NO";
  return "YES";
}
let result = gameOfThrones(s);
console.log(result);
